OPAL/apk/src/main/scala/org/opalj/apk/analyses/ContextRegisteredReceiversAnalysis.scala

```diff
@@ -0,0 +1,144 @@
+/* BSD 2-Clause License - see OPAL/LICENSE for details. */
+package org.opalj.apk.analyses
+
+import org.opalj.apk.ApkContextRegisteredReceiver
+import org.opalj.br.ObjectType
+import org.opalj.br.analyses.Project
+import org.opalj.br.instructions.MethodInvocationInstruction
+import org.opalj.tac.{AITACode, DUVar, LazyDetachedTACAIKey, Stmt, TACMethodParameter}
+import org.opalj.value.{TheStringValue, ValueInformation}
+
+import scala.collection.mutable.ListBuffer
+
+/**
+ * Analyzes code of an APK for dynamically registered Intents for Broadcast Receivers.
+ *
+ * @author Nicolas Gross
+ */
+object ContextRegisteredReceiversAnalysis {
+    val RegisterReceiverMethod = "registerReceiver"
+    val ContextClass = "android.content.Context"
+    val LocalBroadcastManagerClass = "androidx.localbroadcastmanager.content.LocalBroadcastManager"
+    val ActivityClass = "android.app.Activity"
+    val IntentFilterClass = "android.content.IntentFilter"
+
+    def analyze(project: Project[_]): Seq[ApkContextRegisteredReceiver] = {
+        val foundReceivers: ListBuffer[ApkContextRegisteredReceiver] = ListBuffer.empty
+
+        // calls from java code
+        val tacProvider = project.get(LazyDetachedTACAIKey)
+        project.allClassFiles.foreach(_.methodsWithBody.foreach(m => {
+            var alreadyFoundCall = false
+            m.body.get.instructionIterator.foreach {
+                case i: MethodInvocationInstruction =>
+                    if (!alreadyFoundCall && i.name.equals(RegisterReceiverMethod) &&
+                        classHierarchyMatches(project, i.declaringClass.mostPreciseObjectType)) {
+                        // potential further calls are found in TAC, remaining bytecode instructions must not be analyzed
+                        alreadyFoundCall = true
+
+                        // only create TAC for methods where 'registerReceiver' was found in bytecode
+                        val tacMethod = tacProvider.apply(m)
+                        tacMethod.stmts.foreach(s => {
+                            val call = if (s.isAssignment && s.asAssignment.expr.isVirtualFunctionCall) {
```

Comment:
Should probably be a `match`. Might be able to use the `VirtualFunctionCallStatement` unapply